media-gfx/dmtx-utils/dmtx_utils-0.7.4.recipe

```diff
@@ -0,0 +1,64 @@
+SUMMARY="A software for reading and writing Data Matrix barcodes"
+DESCRIPTION="This is open source software for reading and writing Data Matrix \
+barcodes on Linux, Unix, OS X, Windows, and mobile devices. At its core \
+libdmtx is a native shared library, allowing C/C++ programs to use \
+its capabilities without extra restrictions or overhead."
+HOMEPAGE="http://libdmtx.sourceforge.net/"
+COPYRIGHT="2009-2017 Mike Laughton and others"
+LICENSE="GNU LGPL v2"
+REVISION="1"
+SOURCE_URI="https://downloads.sourceforge.net/libdmtx/dmtx-utils-$portVersion.tar.gz"
+CHECKSUM_SHA256="27104dba2a53979c30a0788f4c9ccbf5e30fbe382b9bb6942928f9f869f528b6"
+SOURCE_DIR="dmtx-utils-$portVersion"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+if [ "$targetArchitecture" != x86_gcc2  ]; then
+	commandBinDir=$binDir
+else
+	commandBinDir=$prefix/bin
+fi
+
+PROVIDES="
+	dmtx_utils$secondaryArchSuffix = $portVersion
+	cmd:dmtxquery = $portVersion
+	cmd:dmtxread = $portVersion
+	cmd:dmtxwrite = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libdmtx$secondaryArchSuffix
+	lib:libgomp$secondaryArchSuffix
```

Comment:
Please replace
```
REQUIRES="
	haiku$secondaryArchSuffix
	lib:libdmtx$secondaryArchSuffix
	lib:libgomp$secondaryArchSuffix
	lib:libMagickCore_6.Q16$secondaryArchSuffix
	lib:libMagickWand_6.Q16$secondaryArchSuffix
	"
```
by
```
REQUIRES="
	haiku
	lib:libdmtx
	lib:libMagickCore_6.Q16
	lib:libMagickWand_6.Q16
	"
if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
	REQUIRES="$REQUIRES
		lib:libgomp
		"
fi
```
This is because x86_gcc2 does not have a libgomp.

Please also get rid of all the other occurences of $secondaryArchSuffix or ${secondaryArchSuffix}.

BTW, we could use $targetArchitecture instead of $effectiveTargetArchitecture because when haikuporter is building a recipe for a primary architecture $targetArchitecture and $effectiveTargetArchitecture are equivalent.
But it is better to use $effectiveTargetArchitecture to make this libgomp "selection trick" consistent with across all recipes that have it, without having to test whether we(re building for a primary or secondary arch.